docs/content/en/docs/implementing/tasks/_index.md

```diff
@@ -0,0 +1,179 @@
+
+- inline
+- referring to an HTTP script
+- referring to another `KeptnTaskDefinition`
+
+### Context
+
+A Kubernetes context is a set of access parameters
+that contains a Kubernetes cluster, a user, a namespace,
+the application name, workload name, and version.
+For more information, see
+[Configure Access to Multiple Clusters](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/).
+
+You may need to include context information in the `function` code
+included in the YAML file that defines a
+[KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md)
+resource.
+For an example of how to do this, see the
+[keptn-tasks.yaml](https://github.com/keptn-sandbox/klt-on-k3s-with-argocd/blob/main/simplenode-dev/keptn-tasks.yaml)
+file.
+
+A context environment variable is available via `Deno.env.get("CONTEXT")`.
+It can be used like this:
+  
+```javascript
+let context = Deno.env.get("CONTEXT");
+    
+if (context.objectType == "Application") {
+    let application_name = contextdata.appName;
+    let application_version = contextdata.appVersion;
+}       
+        
+if (context.objectType == "Workload") {
+    let application_name = contextdata.appName;
+    let workload_name = contextdata.workloadName;
+    let workload_version = contextdata.workloadVersion;
+}
+```
+
+## Parameterized functions
+
+`KeptnTaskDefinition`s can use input parameters.
+Consider the following example:
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v1alpha2
+kind: KeptnTaskDefinition
+metadata:
+  name: slack-notification-dev
+spec:
+  function:
+    functionRef:
+      name: slack-notification
+    parameters:
+      map:
+        textMessage: "This is my configuration"
+    secureParameters:
+      secret: slack-token
+```
+
+Note the following about using parameters with functions:
+
+- The Lifecycle Toolkit passes the values
+  defined inside the `map` field as a JSON object.
+- Multi-level maps are not currently supported.
+- The JSON object can be read through the environment variable `DATA`
+  using `Deno.env.get("DATA");`.
+- Currently only one secret can be passed.
+  The secret must have a `key` called `SECURE_DATA`.
+  It can be accessed via the environment variable `Deno.env.get("SECURE_DATA")`.
+
+## Create secret text
+
+To create a secret to use in a `KeptnTaskDefinition`,
+execute this command:
+
+```yaml
+# kubectl create secret generic my-secret --from-literal=SECURE_DATA=foo
+  
+apiVersion: lifecycle.keptn.sh/v1alpha3
+kind: KeptnTaskDefinition
+metadata:
+  name: dummy-task
+  namespace: "default"
+spec: 
+  function: 
+    secureParameters:
+      secret: my-secret
+    inline:
+      code: |
+        let secret_text = Deno.env.get("SECURE_DATA");
+        // secret_text = "foo"
+```
+
+This methodology supports multiple variables
+by creating a Kubernetes secret with a JSON string:
+
+```yaml
+# kubectl create secret generic my-secret \
+# --from-literal=SECURE_DATA="{\"foo\": \"bar\", \"foo2\": \"bar2\"}"
```

Comment:
I think the shell command should not be part of the yaml here. Let's try to do it like here: https://github.com/keptn/lifecycle-toolkit/blob/main/README.md?plain=1#L62

Reply:
fixed